recipes/IWSLT22_lowresource/train.py

```diff
@@ -0,0 +1,420 @@
+            dynamic_items=[audio_pipeline, reference_text_pipeline],
+            output_keys=[
+                "id",
+                "sig",
+                "duration",
+                "trans",
+                "tokens_list",
+                "tokens_bos",
+                "tokens_eos",
+            ],
+        )
+
+    # Sorting training data with ascending order makes the code  much
+    # faster  because we minimize zero-padding. In most of the cases, this
+    # does not harm the performance.
+    if hparams["sorting"] == "ascending":
+        if hparams["debug"]:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                key_min_value={"duration": hparams["sorting_min_duration"]},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+                reverse=True,
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                key_min_value={"duration": hparams["sorting_min_duration"]},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+                reverse=True,
+            )
+        else:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                sort_key="duration"
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                sort_key="duration"
+            )
+
+        hparams["dataloader_options"]["shuffle"] = False
+        hparams["dataloader_options"]["shuffle"] = False
+    elif hparams["sorting"] == "descending":
+        # use smaller dataset to debug the model
+        if hparams["debug"]:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                key_min_value={"duration": hparams["sorting_min_duration"]},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+                reverse=True,
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                key_min_value={"duration": hparams["sorting_min_duration"]},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+                reverse=True,
+            )
+        else:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                sort_key="duration", reverse=True
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                sort_key="duration", reverse=True
+            )
+
+        hparams["dataloader_options"]["shuffle"] = False
+        hparams["dataloader_options"]["shuffle"] = False
+    elif hparams["sorting"] == "random":
+        # use smaller dataset to debug the model
+        if hparams["debug"]:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                key_min_value={"duration": 3},
```

Comment:
Going through Titouan's earlier comments and the current version. Here's a hard-coded value that could be read from hparams.